catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -316,28 +324,25 @@ public Response viewRegInfoHtml(@PathParam("metacardId") final String metacardId
     private List<String> getAddresses(List<PostalAddressType> addressTypes) {
         List<String> addresses = new ArrayList<>();
         for (PostalAddressType address : addressTypes) {
-            String city = "";
-            String street = "";
-            String stateOrProvince = "";
-            String postalCode = "";
-            String country = "";
+            ArrayList<String> addressArray = new ArrayList<>();
+
+            if (address.isSetStreet()) {
+                addressArray.add(address.getStreet());
+            }
             if (address.isSetCity()) {
-                city = address.getCity() + " ";
+                addressArray.add(address.getCity());
             }
-            if (address.isSetCountry()) {
-                country = address.getCountry();
+            if (address.isSetStateOrProvince()) {
+                addressArray.add(address.getStateOrProvince());
             }
             if (address.isSetPostalCode()) {
-                postalCode = address.getPostalCode() + " ";
+                addressArray.add(address.getPostalCode());
             }
-            if (address.isSetStateOrProvince()) {
-                stateOrProvince = address.getStateOrProvince() + " ";
-            }
-            if (address.isSetStreet()) {
-                street = address.getStreet() + " ";
+            if (address.isSetCountry()) {
+                addressArray.add(address.getCountry());
             }
 
-            String addressString = street + city + stateOrProvince + postalCode + country;
+            String addressString = StringUtils.join(addressArray, " ");
```

Comment:
If the array is empty nothing will show up. It will only join on spaces for all but the last value, so if it's empty it should just be blank.